src/journey-maps/angular/components/template-renderer/template-renderer.ts

```diff
@@ -0,0 +1,68 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  ElementRef,
+  Input,
+  OnChanges,
+  SimpleChanges,
+  TemplateRef,
+  ViewChild,
+} from '@angular/core';
+
+import { SbbTemplateType } from '../../journey-maps.interfaces';
+
+@Component({
+  selector: 'sbb-template-renderer',
+  templateUrl: './template-renderer.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class SbbTemplateRenderer implements OnChanges {
+  @Input() templateContext: any;
+  @Input() template?: SbbTemplateType;
+
+  ngTemplate?: TemplateRef<any>;
+  htmlTemplate?: HTMLElement;
+
+  private _htmlTemplateContainer?: ElementRef<HTMLElement>;
+  private _htmlTemplateObserver: MutationObserver;
+
+  @ViewChild('htmlTemplateContainer') set htmlTemplateContainer(
+    container: ElementRef<HTMLElement>
+  ) {
+    this._htmlTemplateContainer = container;
+    this.addHtmlTemplateToDOM();
+  }
+
+  ngOnChanges(changes: SimpleChanges): void {
+    if (!changes.template) {
+      return;
+    }
+
+    this._htmlTemplateObserver?.disconnect();
+
+    if (typeof this.template === 'string') {
+      this.ngTemplate = undefined;
+      const content = (document.getElementById(this.template) as HTMLTemplateElement).content;
```

Comment:
This might not work in a project that (extensively) uses Shadow DOM, as it does not pierce Shadow DOM.
But I don't have an alternative suggestion at the moment.

Reply:
Thanks for the input.

Can we leave it like that for the moment and address the issue later?

Reply:
That's fine. 👍